feat: made first 3 screens of activity 1

darren2hang opened the pull request:
- Made TextBubble component that displays cartton speech bubbles
- Made Convo component that can display multiple text bubbles in a row
- fixed EOL to LF
Closes: #11

darren2hang committed:
Try to fix eol to LF

darren2hang committed:
Add TextBubbleStyles enum

darren2hang committed:
Remove duplicate activity1 file

reginawang99 commented on src/assets/lost-in-translation.svg:
```diff
@@ -1,4 +1,4 @@
-<svg width="350" height="270" viewBox="0 0 350 270" fill="none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
```
oop no yarn lint here pls

darren2hang replied:
ok thanks! I removed it. I think I fixed everything

darren2hang committed:
Removed typo

tr89on commented:
from a visual standpoint there's a lot of white space around each slide on my monitor, so not sure if scaling/centering it would be better (subjective)---this might have more to do with how the carousel display works though

![Screen Shot 2021-02-26 at 3 02 26 PM](https://user-images.githubusercontent.com/22901219/109364428-a4ebf080-7843-11eb-9966-79b628cd6acc.png)

other than that the slides look good and the animation is clean 😎

tr89on commented on src/components/Activities/Activity1/TextBubble.tsx:
```diff
@@ -0,0 +1,35 @@
+      {(props.textBubbleStyle === TextBubbleStyles.LARGE) ? <img src={LargeTextBubbleSvg} />: <span></span>}
+      {(props.textBubbleStyle === TextBubbleStyles.MEDIUM) ? <img src={MediumTextBubbleSvg} />: <span></span>}
+      {(props.textBubbleStyle === TextBubbleStyles.SMALL_LEFT) ? <img src={SmallLeftTextBubbleSvg} />: <span></span>}
+      {(props.textBubbleStyle === TextBubbleStyles.SMALL_RIGHT) ? <img src={SmallRightTextBubbleSvg} />: <span></span>}
```
Instead of returning a bunch of empty <span> it might be better to just return '' (empty string) for the unused textbubble styles

darren2hang replied:
That's a really good idea! I didn't think of that.

tr89on commented on src/components/Activities/Activity1/Convo.tsx:
```diff
@@ -0,0 +1,51 @@
+
+  return (
+    <div>
+      {props.phrases.map(displaySpeechBubbles)}
```
this could just be a single function to display current textBubble instead of map--right now it's effectively only displaying 1 bubble at a time (?) since the other list items return empty divs (similar to the empty `<span>`)

darren2hang replied:
Yep, you're right!

reginawang99 commented on src/components/Activities/Activity1/TextBubble.tsx:
```diff
@@ -0,0 +1,35 @@
+  textBubbleStyle: TextBubbleStyles,
+}
+function TextBubble(props: TextBubbleProps): JSX.Element {
+  // processSvg ()
+  return (
+    <div className='text-bubble'>
+      {/* <img src = {} /> */}
+      {(props.textBubbleStyle === TextBubbleStyles.EXTRA_LARGE) ? <img src={ExtraLargeTextBubbleSvg} />: <span></span>}
+      {(props.textBubbleStyle === TextBubbleStyles.LARGE) ? <img src={LargeTextBubbleSvg} />: <span></span>}
+      {(props.textBubbleStyle === TextBubbleStyles.MEDIUM) ? <img src={MediumTextBubbleSvg} />: <span></span>}
+      {(props.textBubbleStyle === TextBubbleStyles.SMALL_LEFT) ? <img src={SmallLeftTextBubbleSvg} />: <span></span>}
+      {(props.textBubbleStyle === TextBubbleStyles.SMALL_RIGHT) ? <img src={SmallRightTextBubbleSvg} />: <span></span>}
```
try to minimize divs/brs/spans in final result unless you have a reason for them
```suggestion
  let image;
  image = (props.textBubbleStyle === TextBubbleStyles.EXTRA_LARGE) ? ExtraLargeTextBubbleSvg : image;
  image = (props.textBubbleStyle === TextBubbleStyles.LARGE) ? LargeTextBubbleSvg : image;
  image = (props.textBubbleStyle === TextBubbleStyles.MEDIUM) ? MediumTextBubbleSvg : image;
  image = (props.textBubbleStyle === TextBubbleStyles.SMALL_LEFT) ? SmallLeftTextBubbleSvg : image;
  image = (props.textBubbleStyle === TextBubbleStyles.SMALL_RIGHT) ? SmallRightTextBubbleSvg : image;

  return (
    <div className='text-bubble'>
      {(!image) ? <></> : <img src={image} />}
```

reginawang99 replied:
also, we are a bit far along so i don't think we need to go back and change it, but take a look at [this commit,](https://github.com/uclaacm/Playnet/commit/346638eb72dc4fe1de5e0e34093fec9f7bc49051) it uses id to specify background image rather than doing all these ternary operations

BryanPan342 replied:
I would recommend the div method because it becomes a lot more readable!

reginawang99 replied:
@BryanPan342 div method? as in the original? or

BryanPan342 replied:
like using a div element and having the background image be associated with an id!

reginawang99 replied:
ya sounds good to me! in fact, this would remove the funny delayed rendering thing you had @darren2hang if you specify width and height for each image! hmu if you have more questions :)

darren2hang replied:
ok, i changed it to use background id.

reginawang99 commented on src/components/Activities/Activity1/index.tsx:
```diff
@@ -0,0 +1,70 @@
+    {
+      topText: 'Computers have to learn just like we do: by trial and error!',
+      child:
+        <div id='slide-3'>
```
nit (not important, feel free to dismiss): make this its own functional component (possibly in same file or not) since its a bit long and clogs up the content area

reginawang3495 reviewed (changes requested):
Looks great! thanks for doing this! :) just some changes to help the code be more easy to read+reuse

BryanPan342 commented on src/components/Activities/Activity1/Convo.tsx:
```diff
@@ -0,0 +1,51 @@
+        </div>;
+      }
+
+      return <div className='fade' style={{animationDuration: 2/3*props.delayTime/1000+'s'}}>
```
@reginawang99  we can switch these all to anime.js! it makes doing all of these animations 10x easier :)

I'll make a PR this weekend to showcase its p0wer

BryanPan342 commented on src/assets/sending-videos.svg:
```diff
@@ -1,3 +1,4 @@
+
```
remove this unnecessary white space!

darren2hang replied:
Ok!

BryanPan342 commented on src/components/Activities/Activity1/Convo.tsx:
```diff
@@ -0,0 +1,51 @@
+  const [i, setI] = useState(0);
+
+  const displaySpeechBubbles = (text: string) => {
+    if (props.phrases[i] === text) {
+      if (i === 0 && props.offset !== undefined && props.phrases[i] === ''){
```
in general, nested conditional statements leads to less readable code

I guess, one of my questions is.. when would `props.phrases[i]` ever not equal `text` (which is just props.phrases[i]...)

darren2hang replied:
This was a workaround because I couldn't think of a way to make a offset time before starting the display of phrases. So i just made the first phrase empty to add an offset time. I just removed this though.

BryanPan342 reviewed (changes requested):
I havent done a thorough review yet, because you have two reviews already, BUT congrats on your first PR for `Playnet`

I'll add more of my thoughts as your resolve these other reviews 😊 Keep at it!

darren2hang committed:
Return empty string instead of empty span in TextBubble

darren2hang committed:
Removed map and replaced with a function that returns a single div

darren2hang committed:
Remove unnecessary white space

darren2hang commented on src/components/Activities/Activity1/Convo.tsx:
```diff
@@ -0,0 +1,51 @@
+import '../../styles/Activity1.scss';
+import { TextBubbleStyles } from '../../shared/PlaynetConstants';
+import TextBubble from './TextBubble';
+
```
like a different delay/pause time for each word? That would actually be pretty useful for it there were longer words in the convo

reginawang99 replied:
yup! maybe it can be defaulted to timeBtwnPhrases if not specified

darren2hang replied:
I changed it to an array of Phrases I didn't know you could accept suggestions through github so I didn't accept the commit suggestion. Sorry lol.

darren2hang committed:
Replaced ternary operators with background id

darren2hang committed:
Offset in Convo now works without requiring an empty string to be passed in

darren2hang committed:
Fix linter errors

darren2hang committed:
Fix linter errors

darren2hang commented on src/components/Activities/Activity1/Convo.tsx:
```diff
@@ -0,0 +1,51 @@
+  };
+
+  useEffect(() => {
+    if (i === 0 && props.offset !== undefined && props.phrases[i] === '')
```
i removed this and reimplemented how the offset works.

reginawang99 commented on src/components/Activities/Activity1/TextBubble.tsx:
```diff
@@ -0,0 +1,35 @@
+
+interface TextBubbleProps {
+  text?: string;
+  contentSvg?: string;   //if broken change back to any
```
@BryanPan342 what do u think, should we do `any` for the images , or have an enum that specifies image, or ? i think string is p misleading

BryanPan342 replied:
im gonna need to review the entire thing before I make a judgement, but just by the looks of it, maybe we should name it something like `svgPath` or smth like that?

darren2hang committed:
Change Convo props to array of Phrase objects

darren2hang commented on src/components/Activities/Activity1/Convo.tsx:
```diff
@@ -0,0 +1,51 @@
+import TextBubble from './TextBubble';
+
+interface ConvoProps {
+  phrases: string[],
```
i just made this change!

darren2hang replied:
i just made this change!

darren2hang commented on src/components/Activities/Activity1/Convo.tsx:
```diff
@@ -0,0 +1,51 @@
+      }
+
+      return <div className='fade' style={{animationDuration: 2/3*props.delayTime/1000+'s'}}>
+        {(props.phrases[i].indexOf('.svg') === -1) ?
```
I just made this change!

darren2hang committed:
Moved slide 3 content to its own functional component to reduce clutter

darren2hang commented:
> from a visual standpoint there's a lot of white space around each slide on my monitor, so not sure if scaling/centering it would be better (subjective)---this might have more to do with how the carousel display works though
> 
> ![Screen Shot 2021-02-26 at 3 02 26 PM](https://user-images.githubusercontent.com/22901219/109364428-a4ebf080-7843-11eb-9966-79b628cd6acc.png)
> 
> other than that the slides look good and the animation is clean 😎

I think regina said she would address this in the carousel content.

reginawang99 commented on src/components/Activities/Activity1/Convo.tsx:
```diff
@@ -0,0 +1,66 @@
+    if (props.waitTimeBeforeStart !== undefined && !offsetYet) {
+      setTimeout(() => setOffSetYet(true), props.waitTimeBeforeStart);
+    } else if (i < props.phrases.length - 1) {
+
```
nit: remove extra space
```suggestion
```

reginawang99 commented on src/components/Activities/Activity1/Convo.tsx:
```diff
@@ -0,0 +1,66 @@
+  phrases: Phrase[],
+  textBubbleStyle: TextBubbleStyles,
+  timeBtwnPhrases: number,
+  waitTimeBeforeStart?: number,
```
ah actually looking now, it doesn't seem like this is necessary since it's simply text = '' and bubble style is none😅, i think this might add a bit of unnecessary complexity (like state variable and such); my b for not cathcing it sooner

reginawang99 replied:
actually looking at how it needs a state variable and such, i think it might be unnecessary complexity, my b for not catching sooner 😅 can we make so empty is just TextBubbleStyle.NONE and text = ''? thanks!
```suggestion
```

reginawang99 commented on src/components/Activities/Activity1/Convo.tsx:
```diff
@@ -0,0 +1,66 @@
+  const [offsetYet, setOffSetYet] = useState(false);
+
+  const DisplaySpeechBubbles = () => {
+    if (props.waitTimeBeforeStart !== undefined && !offsetYet) {
```
```suggestion
    // setup empty text bubble 
    if (props.waitTimeBeforeStart !== undefined && !offsetYet) {
```

reginawang99 commented on src/components/Activities/Activity1/Convo.tsx:
```diff
@@ -0,0 +1,66 @@
+  const [offsetYet, setOffSetYet] = useState(false);
+
+  const DisplaySpeechBubbles = () => {
+    if (props.waitTimeBeforeStart !== undefined && !offsetYet) {
+      return <div key={'blank'} className='hidden'>
+        <TextBubble key={i} text={''} textBubbleStyle={TextBubbleStyles.NONE} />
+      </div>;
+    }
+
```
```suggestion

      // setup text bubble with contents
```

reginawang99 replied:
as per above!
```suggestion
```

reginawang3495 reviewed (changes requested):
just a super tiny change 😅 and then im good to go! it looks great! nice and reusable parts :))

reginawang99 commented on src/components/Activities/Activity1/Convo.tsx:
```diff
@@ -0,0 +1,66 @@
+
+function Convo(props: ConvoProps): JSX.Element {  //to replay convo, set i to 0 and offsetYet to false
+  const [i, setI] = useState(0);
+  const [offsetYet, setOffSetYet] = useState(false);
```
```suggestion
```

reginawang99 commented on src/components/Activities/Activity1/Convo.tsx:
```diff
@@ -0,0 +1,66 @@
+    </div>;
+  };
+
+  useEffect(() => {
+    if (props.waitTimeBeforeStart !== undefined && !offsetYet) {
+      setTimeout(() => setOffSetYet(true), props.waitTimeBeforeStart);
+    } else if (i < props.phrases.length - 1) {
```
as per above!
```suggestion
  useEffect(() => {
    if (i < props.phrases.length - 1) {
```

reginawang99 commented on src/components/Activities/Activity1/index.tsx:
```diff
@@ -0,0 +1,98 @@
+    },
+  ];
+  const computerGuesses: Phrase[] = [
+    {
```
as per above!
```suggestion
    {
      text: '',
      isText: true,
      timeOnScreen: timeBtwnWords / 2,
      textBubbleStyle: TextBubbleStyles.NONE,
    },
```

reginawang99 commented on src/components/Activities/Activity1/index.tsx:
```diff
@@ -0,0 +1,98 @@
+        </div>
+        <div className='right-content content'>
+          <div id='computer-speech-bubble'>
+            <Convo phrases={computerGuesses} timeBtwnPhrases={timeBtwnWords} waitTimeBeforeStart={timeBtwnWords / 2}
```
as per above!
```suggestion
            <Convo phrases={computerGuesses} timeBtwnPhrases={timeBtwnWords}
```

reginawang99 commented:
ahh as per the carousel stuffs, im not actually sure whats better, maybe we can also discuss with the designers later!

darren2hang committed:
Remove extra space

Co-authored-by: Regina Wang <rrreggginnna@gmail.com>

darren2hang committed:
Remove waitDelay argument to Convo

Replaced the delay variable before starting the display of textbubbles, with just having an empty text bubble with TextBubbleStyles.NONE

Co-authored-by: Regina Wang <rrreggginnna@gmail.com>

darren2hang committed:
Fix typo in Regina's suggested changes

reginawang3495 reviewed (approved):
yay nice nice nice nice LGTM!! 🎉

BryanPan342 reviewed (approved):
LGTM congrats on ur first PR :)

mergify[bot] committed:
Merge branch 'main' into activty1_fixeEOLattempt3
